lib/spack/spack/test/cmd/buildcache.py

```diff
@@ -848,3 +850,192 @@ def test_buildcache_prune_orphaned_manifest(tmp_path, mutable_database, mock_gnu
+
+    # spec sources should raise an exception
+    expect = "Could not determine spec source type of DEADBEEF"
+    with pytest.raises(spack.error.SpackError, match=expect):
+        command_args = ["create-view", "--name", "test_view", "my-mirror", "DEADBEEF"]
+        buildcache(*command_args)
+
+    invalid_field_json = tmp_path / "invalid_field_json.json"
+    invalid_field_json.write_text('{"DEADBEEF": []}')
+    with pytest.raises(spack.error.SpackError, match=str(invalid_field_json)):
+        command_args = ["create-view", "--name", "test_view", "my-mirror", str(invalid_field_json)]
+        buildcache(*command_args)
+
+
+def test_buildcache_create_view_empty(
+    tmp_path, mutable_config, mutable_database, mutable_mock_env_path
+):
+    mirror_directory = str(tmp_path)
+    mirror("add", "--unsigned", "my-mirror", mirror_directory)
+
+    # Push a spec to the cache
+    mpileaks_specs = mutable_database.query_local("mpileaks")
+    buildcache("push", "--update-index", "my-mirror", mpileaks_specs[0].format("{/hash}"))
+
+    # Make sure the view doesn't exist yet
+    with pytest.raises(spack.binary_distribution.FetchIndexError):
+        hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+
+    # Write a minimal lockfile (this is not validated/read by an enviornment)
+    empty_lockfile = tmp_path / "emptylock" / "spack.lock"
+    empty_lockfile.parent.mkdir()
+    empty_lockfile.write_text(
+        '{"_meta": {"lockfile-version": 1}, "roots": [], "concrete_specs": {}}', encoding="utf-8"
+    )
+    # Create a view with now specs
+    command_args = [
+        "create-view",
+        "--force",
+        "--name",
+        "test_view",
+        "my-mirror",
+        str(empty_lockfile),
+    ]
+    out = buildcache(*command_args)
+    assert "No specs found for view, creating an empty index" in out
+    hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
+    assert not hashes_in_view
+
+
+@pytest.mark.parametrize("sources", ("hash", "specfile", "env", "lockfile"))
+def test_buildcache_create_view(
+    tmp_path, mutable_config, mutable_database, mutable_mock_env_path, sources
+):
+    mirror_directory = str(tmp_path)
+    mirror("add", "--unsigned", "my-mirror", mirror_directory)
+
+    # Push all of the mpileaks packages to the cache
+    mpileaks_specs = mutable_database.query_local("mpileaks")
+    for s in mpileaks_specs:
+        buildcache("push", "--update-index", "my-mirror", s.format("{/hash}"))
+
+    # Grab all of the hashes for each mpileaks spec
+    mpileaks_0_hashes = [s.dag_hash() for s in mpileaks_specs[0].traverse()]
+    mpileaks_1_hashes = [s.dag_hash() for s in mpileaks_specs[1].traverse()]
+    mpileaks_2_hashes = [s.dag_hash() for s in mpileaks_specs[2].traverse()]
+
+    # Make sure the view doesn't exist yet
+    with pytest.raises(spack.binary_distribution.FetchIndexError):
+        hashes_in_view = read_specs_in_index(mirror_directory, "test_view")
```

Comment:
I'm seeing this repeated. Should it be a user function?

Reply:
I agree it probably makes sense to have a way to check if the configuration of a mirror is going to do what the user wants. I kind of have a check that the version of the layout supports features in a mirror, but that is also hidden.

It has been on my wish list to have a `spack mirror check <mirrors...>` which would do a more indepth check of the mirror and report things like orphaned blobs, holes in dependencies, feature compatibility with config, deprecated layouts, signatures match match with keys in key index, etc. with options to run the different checks. Running these types of checks hit the network, so I don't really want to run them unless the user wants them.